src/containers/monitor.jsx

```diff
@@ -68,39 +80,51 @@ class Monitor extends React.Component {
         );
     }
     handleNextMode () {
-        // @todo the mode list needs to be filtered for current available modes
-        // i.e. no sliders for read-only monitors, only list for list vars.
-        const modeIndex = monitorModes.indexOf(this.state.mode);
-        this.setState({mode: monitorModes[(modeIndex + 1) % monitorModes.length]});
+        const modes = availableModes(this.props.opcode);
+        const modeIndex = modes.indexOf(this.state.mode);
+        this.setState({mode: modes[(modeIndex + 1) % modes.length]});
     }
     handleSetModeToDefault () {
         this.setState({mode: 'default'});
     }
     handleSetModeToLarge () {
         this.setState({mode: 'large'});
     }
+    handleSetModeToSlider () {
+        this.setState({mode: 'slider'});
+    }
     setElement (monitorElt) {
         this.element = monitorElt;
     }
     render () {
+        // console.log(this.props);
         const monitorProps = monitorAdapter(this.props);
+        const showSliderOption = availableModes(this.props.opcode).indexOf('slider') !== -1;
         return (
             <MonitorComponent
                 componentRef={this.setElement}
                 {...monitorProps}
+                height={this.props.height}
+                max={this.props.max}
+                min={this.props.min}
                 mode={this.state.mode}
+                width={this.props.width}
                 onDragEnd={this.handleDragEnd}
                 onNextMode={this.handleNextMode}
                 onSetModeToDefault={this.handleSetModeToDefault}
                 onSetModeToLarge={this.handleSetModeToLarge}
+                onSetModeToSlider={showSliderOption ? this.handleSetModeToSlider : null}
```

Comment:
Might be simpler to handle this in the handler and then always pass it in? Then you don't have to conditionally check the type of this prop in the component.